index.js

```diff
@@ -83,6 +99,133 @@ function attachToBody (options, req, reply, next) {
   })
 }
 
+const allowedTokens = ['[', ']', '{}']
+
+function getSerializedKeyTokens (key) {
+  const tokens = key.split(/(\[|]|{})/).filter(Boolean)
+  return tokens.some(token => allowedTokens.includes(token)) ? tokens : []
```

Comment:
.some() is potentially slower than a for loop.

Reply:
Replaced it with a negative while, which was the fastest alternative.